File: app/src/javax/java/com/adawolfs/mvp/mvp/BaseMvpPresenter.java
package com.adawolfs.mvp.mvp;

/**
 * Created by adawolfs on 1/20/17.
 */

public interface BaseMvpPresenter<V extends BaseMvpView> {

    void attachView(V view);

    void detachView();

    V getView();

}

File: app/src/kotlin/java/com/adawolfs/mvp_java/mvp/BaseMvpPresenter.kt
package com.adawolfs.mvp_java.mvp

/**
 * Created by adawolfs on 1/20/17.
 */
interface BaseMvpPresenter<V: BaseMvpView>  {

    fun attachView(view: V)

    fun detachView()


    fun getView(): V?

}